### ISSUE TITLE
sccache error with gh actions cache on windows since 0.5.1

### ISSUE THREAD

**Issue by messense** [mozilla/sccache#1789]
```
sccache: error: failed to execute compile
sccache: caused by: Failed to send data to or receive data from server
sccache: caused by: Failed to read response header
sccache: caused by: An existing connection was forcibly closed by the remote host. (os error 10054)
sccache: error: failed to execute compile
sccache: caused by: Failed to send data to or receive data from server
sccache: caused by: Failed to read response header
sccache: caused by: An existing connection was forcibly closed by the remote host. (os error 10054)
error: could not compile `memchr` (lib)
```

Works fine with 0.5.0, broken on 0.5.1 and 0.5.2.

**Comment by messense** [comment]
Might be related to the opendal version bump in https://github.com/mozilla/sccache/pull/1779, cc @Xuanwo

**Comment by alex** [comment]
I'm seeing this, but only intermittently, it doesn't happen on all builds.

**Comment by Xuanwo** [comment]
Let me take a look.

**Comment by Xuanwo** [comment]
Maybe related to the rustls changes?

**Comment by Xuanwo** [comment]
After reviewing the changes between v0.34.0 and v0.36.0, I did not find any related modifications. However, I am still investigating.

**Comment by Xuanwo** [comment]
Questions:

- Does it only happen on windows?
- Does it only happen on ghac? Is s3 and azblob service affected?

Some progress: https://discord.com/channels/1081052318650339399/1081052319342407715/1114820330066813041 (join in via [link](https://discord.gg/XQy8yGR2dg))

**Comment by sylvestre** [comment]
For me, it is only Windows. I haven't tried with other than ghac

**Comment by Xuanwo** [comment]
I found a strange behavior:

```shell
[0m[38;5;8m[[0m2023-06-04T09:04:51Z [0m[34mDEBUG[0m opendal::services[0m[38;5;8m][0m service=ghac operation=read path=2/3/e/23ebe938a116256142bac808d0e240dcdbda53b5d81474223aaf05dab7205be5 range=0- -> started
[0m[38;5;8m[[0m2023-06-04T09:04:51Z [0m[34mDEBUG[0m reqwest::connect[0m[38;5;8m][0m starting new connection: https://artifactcache.actions.githubusercontent.com/
[0m[38;5;8m[[0m2023-06-04T09:04:51Z [0m[34mDEBUG[0m hyper::client::connect::http[0m[38;5;8m][0m connecting to 13.107.42.16:443
[0m[38;5;8m[[0m2023-06-04T09:05:51Z [0m[34mDEBUG[0m sccache::compiler::compiler[0m[38;5;8m][0m [cfg_if]: Cache timed out 0.000 s
[0m[38;5;8m[[0m2023-06-04T09:05:51Z [0m[34mDEBUG[0m sccache::compiler::compiler[0m[38;5;8m][0m [cfg_if]: Compiling locally
[0m[38;5;8m[[0m2023-06-04T09:05:51Z [0m[34mDEBUG[0m sccache::compiler::compiler[0m[38;5;8m][0m [autocfg]: Cache timed out 0.000 s
```

There is a 60s spent on connecting to ghac.

**Comment by sylvestre** [comment]
oh, maybe some rate limiting ? 
(default timeout of 60 seconds is probably a bit too long but this is a different issue)

**Comment by Xuanwo** [comment]
> oh, maybe some rate limiting ?

However, it cannot explain why version 0.5.0 is functional. I am currently investigating this matter.

**Comment by Xuanwo** [comment]
Thanks to @sylvestre for the 0.5.3 release.

@messense Can you take it a try?

**Comment by messense** [comment]
0.5.3 works for me now.

**Comment by Xuanwo** [comment]
I'll keep this issue open until https://github.com/apache/incubator-opendal/issues/2412 fixed.

**Comment by Xuanwo** [comment]
https://github.com/apache/incubator-opendal/issues/2412 has been fixed, I will release opendal 0.37 and bump sccache after that.
